### PR TITLE
Scan file_put_contents

### LINKED ISSUE #182: if an app implements own upload endpoints the antivirus is not triggered
tested with the text editor:
1. create a new file
2. enter the eicar test virus code: `X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*`

the antivirus app does not get triggered and the virus file gets saved

### PR DESCRIPTION
**Pull request by VicDeo**
Fixes #182 

This PR cancel any attempt to put a virus signature with `file_put_contents` by throwing `OCP\Files\ForbiddenException`.
It's up to the app that called `file_put_contents` to catch `OCP\Files\ForbiddenException` and properly expose it's message.

### CODE REVIEW

File: lib/AvirWrapper.php [hunk 1]
```diff
@@ -144,6 +137,60 @@ function () use ($scanner, $path) {
...
+				'',
+				$owner,
+				Activity::TYPE_VIRUS_DETECTED,
+				Activity::PRIORITY_HIGH
+			);
+
+			if ($shouldDelete){
+				//prevent from going to trashbin
+				if (App::isEnabled('files_trashbin')) {
+					\OCA\Files_Trashbin\Storage::preRenameHook([
```
Comment (PVince81) [inline lib/AvirWrapper.php]: :see-no-evil: (I know, old code)

Comment (PVince81) [reply lib/AvirWrapper.php]: @VicDeo can you make a ticket to look into ways to get rid of this eventually ? I hope this isn't causing side effects in some scenarios.

**Comment by PVince81** [comment]
Ahem, unit tests ?

File: lib/AvirWrapper.php (hunk 1 above)
Comment (VicDeo) [reply lib/AvirWrapper.php]: @PVince81 https://github.com/owncloud/files_antivirus/issues/199

**Comment by VicDeo** [comment]
@PVince81 woohoo. those new tests finally passed :tada:

**Pull request merged**